src/modifiers.ts

```diff
@@ -90,249 +96,216 @@ function hasSpecialChars(str: string): boolean {
     return !/^[a-zA-Z0-9]+$/.test(str);
 }
 
+const getPatternToMatch = (selector: string | undefined): RegExp | undefined => {
+    return selector
+        ? !hasSpecialChars(selector)
+            ? new RegExp(`^${selector}$`, 'gi')
+            : new RegExp(selector, 'gi')
+        : undefined;
+};
+
 export class Modifiers {
     codeModel: CodeModel;
-    allCommandGroups: string[];
+    allCommandGroups: Record<string, number>;
+    groupChanged: boolean;
 
     constructor(protected session: Session<CodeModel>) {
         this.codeModel = session.model;
-        this.getAllCommandGroup();
+        this.initAllCommandGroup();
+        this.groupChanged = false;
+    }
+
+    initAllCommandGroup(): void {
+        this.allCommandGroups = {};
+        let groupIdx = 0;
+        this.codeModel.operationGroups.forEach((operationGroup) => {
+            this.allCommandGroups[operationGroup.language['az'].command] = groupIdx;
+            groupIdx++;
+        });
+    }
+
+    paramModify(parameter: Parameter, directive: any): void {
+        const parameterRegex = getPatternToMatch(directive.where['parameter-name']);
+        const parameterReplacer =
+            directive.set !== undefined ? directive.set['parameter-name'] : undefined;
+        const paramDescriptionReplacer =
+            directive.set !== undefined ? directive.set['parameter-description'] : undefined;
+        if (
+            !isNullOrUndefined(parameter.language['az'].name) &&
+            !isNullOrUndefined(parameterRegex) &&
+            parameter.language['az'].name.match(parameterRegex)
+        ) {
+            parameter.language['az'].name = parameterReplacer
+                ? parameterRegex
+                    ? parameter.language['az'].name.replace(parameterRegex, parameterReplacer)
+                    : parameterReplacer
+                : parameter.language['az'].name;
+            parameter.language['az'].mapsto = parameter.language['az'].name.replace(/-/g, '_');
+            parameter.language['az'].description =
+                paramDescriptionReplacer || parameter.language['az'].description;
+        }
     }
 
-    getAllCommandGroup() {
-        this.allCommandGroups = [];
-        for (const group of values(this.codeModel.operationGroups)) {
-            this.allCommandGroups.push(group.language['az'].command);
+    operationGroupModify(operationGroup: OperationGroup, directive: any, groupIdx: number): void {
```

Comment:
modifyOperationGroup

Reply:
changed, and I also changed operationModify => modifyOperation, paramModify => modifyParam